lib/emarsys/client.rb

```diff
@@ -12,25 +12,25 @@ def password
     end
 
     def x_wsse_string
-      string = 'UsernameToken '
-      string += 'Username="' + username + '", '
-      string += 'PasswordDigest="' + header_password_digest + '", '
-      string += 'Nonce="' + header_nonce + '", '
-      string += 'Created="' + header_created + '"'
-      string
+      str = <<-STRING
+      UsernameToken Username="#{username}", PasswordDigest="#{header_password_digest}", Nonce="#{header_nonce}", Created="#{header_created}"
+      STRING
+      str.strip
     end
 
     def header_password_digest
       Base64.encode64(calculated_digest).gsub("\n", "")
     end
 
     def header_nonce
-      bytes = Random::DEFAULT.bytes(16)
-      bytes.each_byte.map { |b| sprintf("%02X",b) }.join
+      @header_nonce ||= begin
```

Comment:
There is already an open PR (#15) to address this issue. Let's focus on the country feature for the time being.